Batteries/Data/Fin/Lemmas.lean

```diff
@@ -47,42 +51,42 @@ theorem foldr_assoc {op : α → α → α} [ha : Std.Associative op] {f : Fin n
 @[simp] theorem findSome?_one {f : Fin 1 → Option α} : findSome? f = f 0 := rfl
 
 theorem findSome?_succ {f : Fin (n+1) → Option α} :
-    findSome? f = (f 0).or (findSome? fun i => f i.succ) := by
+    findSome? f = (f 0).or (findSome? (f ·.succ)) := by
   simp only [findSome?, foldl_succ, Option.orElse_eq_orElse, Option.orElse_eq_or]
   exact Eq.trans (by cases (f 0) <;> rfl) foldl_assoc
 
+@[deprecated findSome?_succ (since := "2025-10-06")]
 theorem findSome?_succ_of_some {f : Fin (n+1) → Option α} (h : f 0 = some x) :
-    findSome? f = some x := by simp [findSome?_succ, h]
+    findSome? f = some x := findSome?_succ.trans (h ▸ Option.some_or)
 
+@[deprecated findSome?_succ (since := "2025-10-06")]
 theorem findSome?_succ_of_isSome {f : Fin (n+1) → Option α} (h : (f 0).isSome) :
-    findSome? f = f 0 := by cases _h : f 0 <;> simp_all [findSome?_succ_of_some]
+    findSome? f = f 0 := findSome?_succ.trans (Option.or_of_isSome h)
 
+@[deprecated findSome?_succ (since := "2025-10-06")]
 theorem findSome?_succ_of_none {f : Fin (n+1) → Option α} (h : f 0 = none) :
-    findSome? f = findSome? fun i => f i.succ := by simp [findSome?_succ, h]
+    findSome? f = findSome? (f ·.succ) := findSome?_succ.trans (Option.or_eq_right_of_none h)
 
+@[deprecated findSome?_succ (since := "2025-10-06")]
```

Comment:
The replacement is not an exact replacement so I don't see the need to deprecate.